### PR TITLE
[IMPROVE] Replace livechat:officeHour publication to REST

### PR DESCRIPTION
**Pull request by MarcosSpessatto**

### CODE REVIEW

File: app/livechat/client/views/app/livechatOfficeHours.js [hunk 1]
```diff
@@ -7,12 +7,12 @@ import moment from 'moment';
 
 import { t, handleError } from '../../../../utils';
 import { settings } from '../../../../settings';
-import { LivechatOfficeHour } from '../../collections/livechatOfficeHour';
 import './livechatOfficeHours.html';
+import { APIClient } from '../../../../utils/client';
```
Comment (sampaiodiego) [inline app/livechat/client/views/app/livechatOfficeHours.js]: move this to other import from `../../../../utils`

File: app/livechat/client/views/app/livechatOfficeHours.js [hunk 2]
```diff
@@ -156,17 +156,16 @@ Template.livechatOfficeHours.onCreated(function() {
...
-			LivechatOfficeHour.find().forEach(function(d) {
-				Template.instance().dayVars[d.day].start.set(moment.utc(d.start, 'HH:mm').local().format('HH:mm'));
-				Template.instance().dayVars[d.day].finish.set(moment.utc(d.finish, 'HH:mm').local().format('HH:mm'));
-				Template.instance().dayVars[d.day].open.set(d.open);
-			});
-		}
+	this.autorun(async () => {
+		const { officeHours } = await APIClient.v1.get('livechat/office-hours');
+		this.officeHours.set(officeHours);
+		this.officeHours.get().forEach((d) => {
```
Comment (sampaiodiego) [inline app/livechat/client/views/app/livechatOfficeHours.js]:
```suggestion
		officeHours.forEach((d) => {
```

File: app/livechat/server/publications/livechatOfficeHours.js [hunk 3]
```diff
@@ -4,8 +4,9 @@ import { hasPermission } from '../../../authorization';
 import { LivechatOfficeHour } from '../../../models';
 
 Meteor.publish('livechat:officeHour', function() {
+	console.warn('The publication "livechat:officeHour" is deprecated and will be removed after version v3.0.0');
```
Comment (sampaiodiego) [inline app/livechat/server/publications/livechatOfficeHours.js]: please move to outside publish as per our discussions

File: app/livechat/client/views/app/livechatOfficeHours.js [hunk 4]
```diff
@@ -156,17 +156,16 @@ Template.livechatOfficeHours.onCreated(function() {
...
-		this.subscribe('livechat:officeHour');
-
-		if (this.subscriptionsReady()) {
-			LivechatOfficeHour.find().forEach(function(d) {
-				Template.instance().dayVars[d.day].start.set(moment.utc(d.start, 'HH:mm').local().format('HH:mm'));
-				Template.instance().dayVars[d.day].finish.set(moment.utc(d.finish, 'HH:mm').local().format('HH:mm'));
-				Template.instance().dayVars[d.day].open.set(d.open);
-			});
-		}
+	this.autorun(async () => {
```
Comment (sampaiodiego) [inline app/livechat/client/views/app/livechatOfficeHours.js]: this doesn't need to be in a `autorun` anymore.. there is no reactivity anymore.

**Comment by renatobecker** [comment]
@MarcosSpessatto can you please fix the conflicts?
Thanks.

**Comment by renatobecker** [comment]
@MarcosSpessatto I just noticed an error on my browser console, as you can see below:

Screen Shot 2019-12-13 at 12 29 53

Screen Shot 2019-12-13 at 12 30 01

File: app/livechat/server/api/lib/officeHour.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { hasPermissionAsync } from '../../../../authorization/server/functions/hasPermission';
+import { LivechatOfficeHour } from '../../../../models/server/raw';
+
+export async function findLivechatOfficeHours({ userId }) {
+	if (!await hasPermissionAsync(userId, 'view-l-room')) {
```
Comment (renatobecker) [inline app/livechat/server/api/lib/officeHour.js]: We have a specific permission for this feature, why are you not using it?
- `view-livechat-officeHours`

Comment (MarcosSpessatto) [reply app/livechat/server/api/lib/officeHour.js]: I used the same permission that was in the publication.

**Pull request merged**